pkg/controller/knativeserving/common/helpers.go

```diff
@@ -19,3 +24,26 @@ func UpdateConfigMap(cm *unstructured.Unstructured, data map[string]string, log
 		unstructured.SetNestedField(cm.Object, v, "data", k)
 	}
 }
+
+// Returns true if we are running in OpenShift
+func IsOpenShift(c client.Client) bool {
+	routeExists, err := AnyKindExists(c, "", schema.GroupVersionKind{"route.openshift.io", "v1", "route"})
+	if err != nil {
+		return false
+	}
+	return routeExists
+}
+
+// AnyKindExists returns true if any of the gvks (GroupVersionKind) exist
+func AnyKindExists(c client.Client, namespace string, gvks ...schema.GroupVersionKind) (bool, error) {
+	for _, gvk := range gvks {
+		list := &unstructured.UnstructuredList{}
+		list.SetGroupVersionKind(gvk)
+		if err := c.List(context.TODO(), &client.ListOptions{Namespace: namespace}, list); err != nil && meta.IsNoMatchError(err) {
+			return false, err
+		}
+		log.Info("Detected", "gvk", gvk.String())
+		return true, nil
+	}
+	return false, nil
+}
```

Comment:
You're gonna hate me for this, but instead of bringing these functions over from `openshift.go` let's instead move all the member roll validation logic out of `checkDependencies` and into a `PreInstall` function in `openshift.go`. What do you think? That way, the extensions logic will do all this for us, and it'll keep the controller clean.